### PR TITLE
update historical averages

### PR DESCRIPTION
**Pull request by kaushikravikumar**
### Summary 

This pull request implements the logic for updating the fitness historical averages every time that live data is inputted from the web app using the submit button. This is done by hitting a POST request endpoint, update-live-averages. 

### Test Plan 

In order to test this we can try submitting data at a time 12:15am, which is a time that facility's are closed. This way, we will not be tampering with any real live average data. Then we simply have to check the firebase database to see if the data is accurately entered into the database.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ module.exports = {
   },
   extends: ["plugin:vue/essential", "eslint:recommended", "@vue/typescript"],
   rules: {
-    "no-console": process.env.NODE_ENV === "production" ? "error" : "off",
```
Comment (SamChou19815) [inline .eslintrc.js:8]: why this change

File: src/views/Home.vue [hunk 2]
```diff
@@ -333,10 +343,48 @@ export default class Home extends Vue {
...
+        const postData =
+        {
+          time: this.roundToQuarter(this.time),
+          cardio: this.cardio,
+          weights: this.weights
+        }
+        const options = {
+          headers: {'Content-Type': 'application/json'}
+        }
+        let url = 'http://localhost:5000/updateLiveAverages' + '?id=' + current_gym + '&day=' + day;
```
Comment (SamChou19815) [inline src/views/Home.vue:378]: You should setup [proxy](https://cli.vuejs.org/config/#devserver-proxy) instead of doing this

File: .eslintrc.js (hunk 1 above)
Comment (kaushikravikumar) [reply .eslintrc.js:8]: is there a reason for the no console.log rule? I thought it was unnecessary. I wanted to log for successful and unsuccessful api requests

Comment (SamChou19815) [reply .eslintrc.js:8]: For frontend code, logging is usually a bad idea since anyone can directly see what you logged. In the past, I see random `console.log` everywhere in the repos where this rule is disabled. It makes debugging harder since you need to find what you logged in a sea of other logs.
With respect to error handling, you should actually deal with the error instead of making it silently fail in the log. Evan making it crash is sometimes better than logging because it makes the problem more visible.

File: src/views/Home.vue [hunk 3]
```diff
@@ -302,14 +303,23 @@ export default class Home extends Vue {
...
   }
 
+  roundToQuarter(time: Date) {
+    const hours = time.getHours()
+    const suffix = (hours >= 12) ? 'pm' : 'am'
+    const hoursRegular = (hours !== 0) ? hours % 12 : 12
+    const minutes = time.getMinutes()
+    const nearestQuarter = (minutes <= 30) ? 15 : 45
+
+    return `${hoursRegular}:${nearestQuarter}${suffix}`
```
Comment (kungpaogao) [inline src/views/Home.vue:313]: Use [Moment.js](https://momentjs.com/)

- https://momentjs.com/docs/#/plugins/round/

File: src/views/Home.vue [hunk 4]
```diff
@@ -333,10 +343,48 @@ export default class Home extends Vue {
...
+          // SUCCESSFUL POST REQUEST :)
+        }).catch(err => {
+          throw err
+        })
+        this.confirm = "";
     } else {
-      this.error = "Please enter a value";
-      return;
+        this.error = "Please enter a value";
+        return;
```
Comment (kungpaogao) [inline src/views/Home.vue:387]: Might need to install [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or something in VS Code

File: src/views/Home.vue [hunk 5]
```diff
@@ -333,10 +343,48 @@ export default class Home extends Vue {
           this.error = "There was an error in adding the document.";
           return;
         });
-      this.confirm = "";
+        let day = ""
```
Comment (benjamin-shen) [inline src/views/Home.vue:346]: `const day = moment(this.time).format("dddd");`

File: src/views/Home.vue [hunk 6]
```diff
@@ -302,14 +303,23 @@ export default class Home extends Vue {
     this.dialog = true;
   }
 
+  roundToQuarter(time: Date) {
+    const hours = time.getHours()
+    const suffix = (hours >= 12) ? 'pm' : 'am'
+    const hoursRegular = (hours !== 0) ? hours % 12 : 12
+    const minutes = time.getMinutes()
+    const nearestQuarter = (minutes <= 30) ? 15 : 45
```
Comment (benjamin-shen) [inline src/views/Home.vue:311]: should this be be `minutes < 30` instead of `<=`?

File: src/views/Home.vue (hunk 3 above)
Comment (benjamin-shen) [reply src/views/Home.vue:313]: Not sure if `moment-round` can round to the nearest `:15`/`:45`.
This is my rounding code from `index.ts` (that does the same thing but returns a moment instead of a string):
```
function roundDate(d: Date) {
  const date = moment(d);
  date.millisecond(Math.floor(date.millisecond() / 1000) * 1000);
  date.second(Math.floor(date.second() / 60) * 60);
  if (round30) {
    date.minute(Math.round((date.minute() + 15) / 30) * 30 - 15);
  } else {
    date.minute(Math.round(date.minute() / 15) * 15);
  }
  return date;
}
```
If you want to return a string, you can `return date.format("h:mma")` instead.
